app/controllers/lectures_controller.rb

```diff
@@ -176,4 +185,26 @@ def validate_course_creator
         redirect_to @course, alert: "You can only access your own courses."
       end
     end
+
+    def generate_enrollment_qr_code
+      enrollment_url = request.base_url + "/courses/" + @course.id.to_s + "/lectures/" + @lecture.id.to_s + "/enroll"
```

Comment:
uuuh, hardcoded stuff inside a controller. i'm sure there must be a rails routes for what you're accessing here